src/InferenceFramework.cpp

```diff
@@ -10,17 +10,17 @@
 namespace EERAModel {
 namespace Inference {
 
-InferenceFramework::InferenceFramework(Model::ModelInterface::Sptr model,
-    const ModelInputParameters& modelInputParameters,
-    const InputObservations& observations,
+InferenceFramework::InferenceFramework(Model::ModelInterface::Sptr& model,
+    ModelInputParameters& modelInputParameters,
+    InputObservations& observations,
     Random::RNGInterface::Sptr rng,
-    const std::string& outDir,
-    Utilities::logging_stream::Sptr log)
+    const std::string outDir,
+    Utilities::logging_stream::Sptr& log)
     : model_(model),
       modelInputParameters_(modelInputParameters),
       observations_(observations),
-      rng_(rng),
-      outDir_(outDir),
+      rng_(std::move(rng)),
```

Comment:
This may be a bad idea. `std:move` will (I think) destroy the original shared pointer. In this case, it doesn't matter, since `rng` isn't used again in `main` after the call to the `InferenceFramework` constructor, but if someone wants to use it in the future after this point in `main`, it's going to cause them problems. In general, I don't think that a call to another module should destroy a shared pointer: the client (in this case `main`) would find that very annoying and a poor design choice.